src/drive/web/modules/public/useSharingInfos.jsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react'
+
+import { useClient, Q } from 'cozy-client'
+
+import logger from 'lib/logger'
+import { getQueryParameter } from 'react-cozy-helpers'
+
+const isShortcutCreatedOnTheRecipientCozy = included => {
+  const sharingMember = included.find(
+    item => item.type === 'io.cozy.sharings.members'
+  )
+  if (sharingMember && sharingMember.attributes.instance) {
+    return true
+  }
+  return false
+}
+
+export const useSharingInfos = () => {
+  const client = useClient()
+
+  const [discoveryLink, setDiscoveryLink] = useState()
+  const [isSharingShortcutCreated, setIsSharingSharingcutCreated] = useState(
+    false
+  )
+  const [sharing, setSharing] = useState()
+  const [loading, setLoading] = useState(true)
+
+  useEffect(
+    () => {
+      const loadSharingDiscoveryLink = async () => {
+        setLoading(true)
+        try {
+          const response = await client
+            .collection('io.cozy.permissions')
+            .getOwnPermissions()
+          const isSharingShortcutCreated = isShortcutCreatedOnTheRecipientCozy(
+            response.included
+          )
+          const sourceId = response.data.attributes.source_id
+          const sharingId = sourceId.split('/')[1]
+          const { sharecode } = getQueryParameter()
+
+          const link = client
+            .collection('io.cozy.sharings')
+            .getDiscoveryLink(sharingId, sharecode)
+          const { data: sharing } = await client.query(
+            Q('io.cozy.sharings').getById(sharingId)
+          )
```

Comment:
THe business logic after the try could be extracted into a fetchDiscoveryLink function ?

Reply:
I don't think so because we do severals things here : 
- fetchPermissions
- check if the shortcutIsCreated
- getDiscoveryLink
- fetchSharing 

However, I can extract the method the get the sharingId from the fetch permissions call response 👍